lib/sidekiq_publisher/publisher.rb

```diff
@@ -44,7 +44,7 @@ def publish_batch(batch, items)
       failure_warning(__method__, ex)
     ensure
       published_count = update_jobs_as_published!(batch) if pushed_count.present? && published_count.nil?
-      metrics_reporter.try(:count, "sidekiq_publisher.published", published_count)
+      metrics_reporter.try(:count, "sidekiq_publisher.published", published_count) if published_count.present?
```

Comment:
Could published_count ever be 0?

Reply:
I can't think of a case where it would be 0. I don't think that this `publish_batch` method will ever receive a batch with no items. And even if Sidekiq middleware filtered out all the jobs so that the `pushed_count` is zero, they would still be marked as published so they are not pushed again.